nixos/modules/services/web-apps/mealie.nix

```
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = cfg.package;
+
+  # TODO Fix ingredient quantities NLP recognition
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    package = lib.mkPackageOption pkgs "mealie" { };
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = "Address on which the service should listen.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service.";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf str;
+      };
+      default = {};
+      description = lib.mdDoc ''
+        Configuration of the Mealie service.
+
+        See [the mealie documentation](https://nightly.mealie.io/documentation/getting-started/installation/backend-config/) for available options and default values.
+
+        In addition to the official documentation, you can set {env}`MEALIE_LOG_FILE`.
+      '';
+      example = {
+        ALLOW_SIGNUP = "false";
+      };
+    };
+
+    credentialsFile = lib.mkOption {
+      type = with lib.types; nullOr path;
+      default = null;
+      example = "/run/secrets/mealie-credentials.env";
+      description = ''
+        File containing credentials used in mealie such as {env}`POSTGRES_PASSWORD`
+        or sensitive LDAP options.
+
+        Expects the format of an `EnvironmentFile=`, as described by {manpage}`systemd.exec(5)`.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.mealie = {
+      description = "Mealie, a self hosted recipe manager and meal planner";
+      after = [
+        "network.target"
+        "network-online.target"
+      ];
+      wants = [
+        "network-online.target"
+      ];
+      wantedBy = [
+        "multi-user.target"
+      ];
+
+      environment = {
+        PYTHONPATH = pkg.python.pkgs.makePythonPath ([ pkg ] ++ pkg.propagatedBuildInputs);
+        PRODUCTION = "true";
+        MEALIE_LOG_FILE = "/var/log/mealie/mealie.log";
```

How much logging happens here? Without log rotation this can be annoying. Ideally logging would go into the journal.

---

Actually, the service logs everything into a file (hardcoded path in the code), so not in the journal.
I only set this env var in order to redirect logs into it.

---

Ideally there would be some sort of log rotation going on, so it doesn't accumulate logs forever.

---

Cases like this can usually be worked around by

```suggestion
        MEALIE_LOG_FILE = "/dev/stdout";
```

---

That unfortunately doesn't work,  `nixosTests.mealie` returns "/dev/stdout not found".
Is there some kind of protection for `/dev/` in systemd services ?
Because with `StandardOutput="journal"` that would be the ideal setup for log rotation

---

However, as I removed `LOGGER_FILE` entirely, I also had to remove some admin features that was fetching logs from the file.

For now, let's keep the "dirty" removal of all of it, as stated in https://github.com/mealie-recipes/mealie/pull/3102#discussion_r1476531794, the logging will be refactored soon, so I'll adapt it to the final form then